editor/src/core/model/project-file-utils.ts

```diff
@@ -175,28 +212,48 @@ function isHTMLComponentFromBaseName(baseName: string, imports: Imports): boolea
   }
 }
 
-export function isImportedComponent(jsxElementName: JSXElementName, imports: Imports): boolean {
-  return Object.keys(imports).some((importKey) => {
-    const fromImports = imports[importKey]
-    return (
-      pluck(fromImports.importedFromWithin, 'name').includes(jsxElementName.baseVariable) ||
-      fromImports.importedWithName === jsxElementName.baseVariable
+export function importInfoFromImportDetails(name: JSXElementName, imports: Imports): ImportInfo {
+  const baseVariable = name.baseVariable
+
+  const err = mapDropNulls((pathOrModuleName) => {
```

Comment:
`err` - couldn't think of a name? :grinning:

Also this probably should not be a `mapDropNulls` followed by a `[0]` indexing, though I appreciate this will be messy either way.

How about (using `forEachValue` from `object-utils.ts`):
```
let importInfo: ImportInfo | null = null
forEachValue((importDetail, pathOrModuleName) => {
  if (importInfo == null) {
    const importAlias = importDetail.importedFromWithin.find(
      (fromWithin) => fromWithin.alias === baseVariable,
    )

    if (importAlias != null) {
      importInfo = createImportedFrom(importAlias.alias, importAlias.name, pathOrModuleName)
    } else if (importDetail.importedAs === baseVariable) {
      importInfo = createImportedFrom(importDetail.importedAs, null, pathOrModuleName)
    } else if (importDetail.importedWithName === baseVariable) {
      importInfo = createImportedFrom(importDetail.importedWithName, null, pathOrModuleName)
    }
  }
}, imports)

return importInfo ?? createNotImported()
```

Reply:
> err - couldn't think of a name? 😀

haha I just made it a const for debugging and left it by accident :D 

I actually wanted to bring this up in the last couple of weeks!
```typescript
let importInfo: ImportInfo | null = null
forEachValue((importDetail, pathOrModuleName) => {
```
I really do not like that more and more places in the codebase we are "forced" to use lets and mutate them. Especially if it's a function that is longer than a dozen lines, it just invites people to do the wrong thing. When I was refactoring the dom-walker and previously the inspector hooks, I had to fight with this a lot. The problem is that this `let` is just too inviting for people down the line who want to append some functionality to overwrite the value 30-40 lines later, embedded in an if. It makes code hard to refactor, and I think these are the functions that eventually end up being a 200 line spagetthi over time because small changes are easiest to make the worst possible way.